lib/Mojo/mysql/Native/Database.pm

```diff
@@ -0,0 +1,227 @@
+package Mojo::mysql::Native::Database;
+use Mojo::Base 'Mojo::mysql::Database';
+
+use Mojo::mysql::Connection;
+use Mojo::mysql::Native::Results;
+use Mojo::mysql::Native::Transaction;
+use Mojo::mysql::Util qw(expand_sql parse_url);
+use Scalar::Util 'weaken';
+use Carp 'croak';
+
+has 'connection';
+
+sub DESTROY {
+  my $self = shift;
+  return unless my $c = $self->connection;
+  return unless my $mysql = $self->mysql;
+  $mysql->_enqueue($c);
+}
+
+sub backlog { scalar @{shift->{waiting} || []} }
+
+sub begin {
+  my $self = shift;
+  $self->query('START TRANSACTION');
+  $self->query('SET autocommit=0');
```

Comment:
Yes, it is blocking for now.
This is a minimal implementation compatible with ::DBI::Transaction.
My idea is to make common ::Transaction module that uses query and async support in future release. This will need API changes.